### DISCUSSION TITLE
Shift ticks and guidline on x-Axis

### DISCUSSION THREAD

**Discussion by nicolasthemmer-aaa** [patrykandpatrick/vico#665] [Questions] [answered]
I want to show all temperature data for a date inside a column of a line chart. As we can see in the current graph, this is not the case. The ticks are shown in the middle value of the series while the label is at the beginning. The goal is to show all values from a series from tick to tick. I know this is working as intended but is there a way to configure this? I haven't found a way to do that yet 🤔 

![image](https://github.com/patrykandpatrick/vico/assets/13311585/bd495548-789f-439e-a5f2-6eb8f1134da8)

**Comment by virtualpathum** [comment]
Hi @Salocin808 , I faced a similar situation and couldn't figure out a way to fix it, I ended up doing something like this, It looks dirty, but does the job for my case. I have two charts with different sizes and expanded and non-expanded views.  I added empty spaces to make it look nice. :). Probably @patrickmichalik can suggest a better approach
```
fun getBottomAxisValueFormatter(xAxisData: List<String>, chartType: String, isExpanded: Boolean): AxisValueFormatter<AxisPosition.Horizontal.Bottom> {
    return AxisValueFormatter { x, _, _ ->
        val index = x.toInt()

        val spaceLength = when (chartType) {
            "I" -> if (isExpanded) 24 else 7
            else -> if (isExpanded) 60 else 19
        }

        val space = " ".repeat(spaceLength)
        if (xAxisData.isNotEmpty()) {
             xAxisData[index]+ space
        } else {
            ""
        }
    }
}
```

**Comment by nicolasthemmer-aaa** [comment, marked as answer]
Hey man :) 

I was able fix my issue by using a different to display data. 
Before I was using `getXStep = { 24f },` in my host (each column displays 1 day -> 24 hours)
The xStep was useful because I only wanted to show the labels for each day. You will have the problem explained above though.

Now I am using the `itemPlacer` for my top axis:

```
             itemPlacer = remember {
                    AxisItemPlacer.Horizontal.default(
                        spacing = 24,
                        offset = 12,
                    )
                },
```

Ignore the custom background coloring in this picture:
![image](https://github.com/patrykandpatrick/vico/assets/13311585/470218a0-7da5-49b2-87e6-532b123427f3)

This will achieve the same result but the chart will interpret each value as a tick and will therefore begin to draw at the very beginning of the chart. (I left the ticks enbaled in the image to see the difference). @virtualpathum with offset and spacing you should be able to get your desired result (if I understood your problem correctly)

I still consider this to be workaround, not exactly a solution to my problem but I am happy :+1:
